### PR TITLE
Feature/10 create collection

### PR DESCRIPTION
**Pull request by SiboYang**
# Summary

Created server and controller files for creating a collection

## Test Plan

Test to be added soon

## Related Issues

#10

### CODE REVIEW

File: backend/src/controllers/muralcollection.controller.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import { EmptyResultError } from "sequelize";
+import { MuralCollectionInterface } from "../models/muralcollection.model";
+import { MuralCollectionService } from "../services/muralcollection.service";
+
+export class MuralCollectionController {
+  public collectionService: MuralCollectionService = new MuralCollectionService();
+
+  // POST /collection
+  public async create(req: Request, res: Response) {
+    const collection: MuralCollectionInterface = req.body.tour;
```
Comment (tedspare) [inline backend/src/controllers/muralcollection.controller.ts:11]: How come `.tour` is used here?

File: backend/src/controllers/muralcollection.controller.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+  //PUT /collection/:id (update a collection by id)
+  public async update(req: Request, res: Response) {
+    const collectionId: number = Number(req.params.id);
+    const params: MuralCollectionInterface = req.body;
+
+    try {
+      await this.collectionService.update(collectionId, params);
+      res.status(202).json({ data: "successfully updated" });
+    } catch (e) {
```
Comment (tedspare) [inline backend/src/controllers/muralcollection.controller.ts:48]: Nice!

File: backend/src/services/muralcollection.service.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+export class MuralCollectionService {
+  public async create(collecton: MuralCollectionInterface, murals: number[]) {
+    const createdCollection: MuralCollection = await MuralCollection.create<MuralCollection>(
+      collecton
+    );
+    murals.forEach(async (muralId) => {
+      const mural = await Mural.findByPk<Mural>(muralId, {
+        rejectOnEmpty: true,
+      });
+      createdCollection.addMural(mural);
```
Comment (tedspare) [inline backend/src/services/muralcollection.service.ts:17]: Nice use of the Postgres association methods!

**Review by tedspare — APPROVED** [review]
Added one comment, otherwise LGTM

File: backend/src/controllers/muralcollection.controller.ts (hunk 1 above)
Comment (SiboYang) [reply backend/src/controllers/muralcollection.controller.ts:11]: omg, I was thinking about tour class when writing it, changing rn!

File: backend/src/services/muralcollection.service.ts (hunk 3 above)
Comment (SiboYang) [reply backend/src/services/muralcollection.service.ts:17]: Thanks!

**Pull request merged**